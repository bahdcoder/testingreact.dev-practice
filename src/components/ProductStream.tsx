import React, { FC } from 'react'
import styled from 'styled-components'
import { Product } from '../types/Product'

import ProductTile from './ProductTile'

interface Props {
  products: Product[]
}

const ProductStream: FC<Props> = ({ products }) => {
  return (
    <Wrapper>
      {products.map((product) => (
        <ProductTile key={product.id} {...product} />
      ))}
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 0 -2.5px;
  > * {
    width: 50%;
    padding: 10px 2.5px 0;
    @media (min-width: 768px) {
      width: 33%;
    }
    @media (min-width: 1024px) {
      width: 25%;
    }
  }
`

export default ProductStream
