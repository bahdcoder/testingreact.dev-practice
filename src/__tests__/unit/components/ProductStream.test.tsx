import React from 'react'
import { render } from '@testing-library/react'
import ProductStream from '../../../components/ProductStream'

describe('The <ProductStream /> component', () => {
    const defaultProducts = [{
        id: 12,
        image: 'image.png',
        price: 'from $12.99',
        brand: 'Adidas',
        createdAt: '2020-02-11 00:00:00',
        isActive: true,
    }, {
        id: 13,
        image: 'image-1.png',
        price: 'from $42.99',
        brand: 'Nike',
        createdAt: '2020-02-11 00:00:00',
        isActive: true,
    }]

    it('renders a list of Product tiles for each product passed to it', async () => {
        const { asFragment, findAllByTestId } = render(
            <ProductStream products={defaultProducts as Array<any>} />
        )

        expect(asFragment()).toMatchSnapshot()
        expect((await findAllByTestId('ProductTile'))).toHaveLength(defaultProducts.length)
    })
})
