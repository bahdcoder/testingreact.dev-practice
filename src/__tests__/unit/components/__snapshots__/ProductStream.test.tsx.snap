// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <ProductStream /> component renders a list of Product tiles for each product passed to it 1`] = `
<DocumentFragment>
  <div
    class="sc-AxheI eOAkpz"
  >
    <div
      class="sc-AxjAm iPShpD"
      data-testid="ProductTile"
    >
      <div
        class="sc-AxirZ fUTalh"
      >
        <img
          alt="Adidas 1"
          class="sc-AxiKw cShrNR"
          data-testid="ProductTileImage"
          src="http://localhost:3443./image.png"
        />
      </div>
      <div
        class="sc-AxgMl dtvmCV"
      >
        <div
          class="sc-AxhCb iMwhLL"
        >
          Adidas 1
        </div>
        <div
          class="sc-AxhUy dqKFMC"
        >
          from $12.99
        </div>
      </div>
    </div>
    <div
      class="sc-AxjAm iPShpD"
      data-testid="ProductTile"
    >
      <div
        class="sc-AxirZ fUTalh"
      >
        <img
          alt="Nike 1"
          class="sc-AxiKw cShrNR"
          data-testid="ProductTileImage"
          src="http://localhost:3443./image-1.png"
        />
      </div>
      <div
        class="sc-AxgMl dtvmCV"
      >
        <div
          class="sc-AxhCb iMwhLL"
        >
          Nike 1
        </div>
        <div
          class="sc-AxhUy dqKFMC"
        >
          from $42.99
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
